From 4261bedd47ad0883332a67858725af44b6741859 Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Tue, 6 Jun 2017 15:14:42 +0800
Subject: [PATCH 1878/5242] MLK-15031-04 tty: serial: fsl_lpuart: wait baud
 rate stable

commit  92765c0e3609e7ababc73cc640435e02fb94178b from
https://source.codeaurora.org/external/imx/linux-imx.git

After set the new baud rate, wait some time for the stable.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Reviewed-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/fsl_lpuart.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/tty/serial/fsl_lpuart.c b/drivers/tty/serial/fsl_lpuart.c
index 43611db..357a576 100644
--- a/drivers/tty/serial/fsl_lpuart.c
+++ b/drivers/tty/serial/fsl_lpuart.c
@@ -12,6 +12,7 @@
 
 #include <linux/clk.h>
 #include <linux/console.h>
+#include <linux/delay.h>
 #include <linux/dma-mapping.h>
 #include <linux/dmaengine.h>
 #include <linux/dmapool.h>
@@ -1894,6 +1895,9 @@ static void lpuart32_shutdown(struct uart_port *port)
 	lpuart32_write(&sport->port, ctrl, UARTCTRL);
 
 	spin_unlock_irqrestore(&sport->port.lock, flags);
+
+	/* wait baud rate stable */
+	usleep_range(1000, 2000);
 }
 
 static const char *lpuart_type(struct uart_port *port)
-- 
1.7.9.5

